File: CSharp/ImageProcessing/Views/FormSettingImageProcessing.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;

namespace ImageProcessing
{
    /// <summary>
    /// FormSettingImageProcessing のロジック
    /// </summary>
    public partial class FormSettingImageProcessing : Form
    {
        /// <summary>
        /// 画像処理タイプのコンボボックス
        /// </summary>
        public System.Windows.Forms.ComboBox CmbBoxImageProcessingType
        {
            get { return cmbBoxImageProcessingType; }
        }

        /// <summary>
        /// コンストラクタ
        /// </summary>
        public FormSettingImageProcessing()
        {
            InitializeComponent();

            LoadParam();
        }

        /// <summary>
        /// デスクトラクタ
        /// </summary>
        ~FormSettingImageProcessing()
        {
        }

        /// <summary>
        /// 設定の読込
        /// </summary>
        public void LoadParam()
        {
            var items = new List<ComImageProcessingType>
            {
                new ComImageProcessingType(Properties.Settings.Default.ImageTypeEdgeId, Properties.Settings.Default.ImageTypeEdgeName),
                new ComImageProcessingType(Properties.Settings.Default.ImageTypeGrayScaleId, Properties.Settings.Default.ImageTypeGrayScaleName),
                new ComImageProcessingType(Properties.Settings.Default.ImageTypeBinarizationId, Properties.Settings.Default.ImageTypeBinarizationName),
                new ComImageProcessingType(Properties.Settings.Default.ImageTypeGrayScale2DiffId, Properties.Settings.Default.ImageTypeGrayScale2DiffName),
                new ComImageProcessingType(Properties.Settings.Default.ImageTypeColorReversalId, Properties.Settings.Default.ImageTypeColorReversalName)
            };

            cmbBoxImageProcessingType.Items.Add(Properties.Settings.Default.ImageTypeEdgeName);
            cmbBoxImageProcessingType.Items.Add(Properties.Settings.Default.ImageTypeGrayScaleName);
            cmbBoxImageProcessingType.Items.Add(Properties.Settings.Default.ImageTypeBinarizationName);
            cmbBoxImageProcessingType.Items.Add(Properties.Settings.Default.ImageTypeGrayScale2DiffName);
            cmbBoxImageProcessingType.Items.Add(Properties.Settings.Default.ImageTypeColorReversalName);
            cmbBoxImageProcessingType.SelectedIndex = (int)items.Find(x => x.Name == Properties.Settings.Default.ImageTypeSelectName)?.Id - 1;

            return;
        }

        /// <summary>
        /// 設定の保存
        /// </summary>
        public void SaveParam()
        {
            Properties.Settings.Default.ImageTypeSelectName = (string)cmbBoxImageProcessingType.SelectedItem;
            Properties.Settings.Default.Save();

            return;
        }

        /// <summary>
        /// OKボタンのクリックイベント
        /// </summary>
        /// <param name="sender">オブジェクト</param>
        /// <param name="e">イベントのデータ</param>
        private void OnClickOk(object sender, EventArgs e)
        {
            SaveParam();
            this.DialogResult = DialogResult.OK;
            Close();
        }

        /// <summary>
        /// Cancelボタンのクリックイベント
        /// </summary>
        /// <param name="sender">オブジェクト</param>
        /// <param name="e">イベントのデータ</param>
        private void OnClickCancel(object sender, EventArgs e)
        {
            this.DialogResult = DialogResult.Cancel;
            Close();
        }
    }
}

File: Vb/ImageProcessing/Views/FormSettingImageProcessing.vb
''' <summary>
''' FormSettingImageProcessing のロジック
''' </summary>
Public Class FormSettingImageProcessing
    ''' <summary>
    ''' 画像処理タイプのコンボボックス
    ''' </summary>
    Public ReadOnly Property ComboBoxImageProcessingType() As System.Windows.Forms.ComboBox
        Get
            Return cmbBoxImageProcessingType
        End Get
    End Property

    ''' <summary>
    ''' コンストラクタ
    ''' </summary>
    Public Sub New()

        ' この呼び出しはデザイナーで必要です。
        InitializeComponent()

        ' InitializeComponent() 呼び出しの後で初期化を追加します。

        LoadParam()
    End Sub

    ''' <summary>
    ''' デスクトラクタ
    ''' </summary>
    Protected Overrides Sub Finalize()
        MyBase.Finalize()
    End Sub

    ''' <summary>
    ''' 設定の読込
    ''' </summary>
    Public Sub LoadParam()
        Dim items = New List(Of ComImageProcessingType)
        items.Add(New ComImageProcessingType(My.Settings.ImageTypeEdgeId, My.Settings.ImageTypeEdgeName))
        items.Add(New ComImageProcessingType(My.Settings.ImageTypeGrayScaleId, My.Settings.ImageTypeGrayScaleName))
        items.Add(New ComImageProcessingType(My.Settings.ImageTypeBinarizationId, My.Settings.ImageTypeBinarizationName))
        items.Add(New ComImageProcessingType(My.Settings.ImageTypeGrayScale2DiffId, My.Settings.ImageTypeGrayScale2DiffName))
        items.Add(New ComImageProcessingType(My.Settings.ImageTypeColorReversalId, My.Settings.ImageTypeColorReversalName))

        cmbBoxImageProcessingType.Items.Add(My.Settings.ImageTypeEdgeName)
        cmbBoxImageProcessingType.Items.Add(My.Settings.ImageTypeGrayScaleName)
        cmbBoxImageProcessingType.Items.Add(My.Settings.ImageTypeBinarizationName)
        cmbBoxImageProcessingType.Items.Add(My.Settings.ImageTypeGrayScale2DiffName)
        cmbBoxImageProcessingType.Items.Add(My.Settings.ImageTypeColorReversalName)
        cmbBoxImageProcessingType.SelectedIndex = items.Find(Function(x) x.Name = My.Settings.ImgTypeSelectName)?.Id - 1

        Return
    End Sub

    ''' <summary>
    ''' 設定の保存
    ''' </summary>
    Public Sub SaveParam()
        My.Settings.ImgTypeSelectName = cmbBoxImageProcessingType.SelectedItem.ToString()
        My.Settings.Save()

        Return
    End Sub

    ''' <summary>
    ''' OKボタンのクリックイベント
    ''' </summary>
    ''' <param name="sender">オブジェクト</param>
    ''' <param name="e">イベントのデータ</param>
    Private Sub OnClickOk(sender As Object, e As EventArgs) Handles btnOk.Click
        SaveParam()
        Me.DialogResult = DialogResult.OK
        Close()
    End Sub

    ''' <summary>
    ''' Cancelボタンのクリックイベント
    ''' </summary>
    ''' <param name="sender">オブジェクト</param>
    ''' <param name="e">イベントのデータ</param>
    Private Sub OnClickCancel(sender As Object, e As EventArgs) Handles btnCancel.Click
        Me.DialogResult = DialogResult.Cancel
        Close()
    End Sub
End Class